packages/react-i18n/src/i18n.ts

```diff
@@ -316,6 +316,45 @@ export class I18n {
     return formatDate(date, locale, {...formatOptions, timeZone});
   }
 
+  /**
+   * Format a Date object to a human readable string, containing both date and time.
+   * It uses the `date.humanize.datetime` format that should be defined in the translations.
+   *
+   * Polaris describes the format that should be used for date/time combinations here:
+   * https://polaris.shopify.com/foundations/content/grammar-and-mechanics#date
+   *
+   * @param {Date} date Date object to format
+   * @param {Intl.DateTimeFormatOptions} [options={}] Options for both date and time formatting
+   * @return {*}  {string} A formatted string
+   * @memberof I18n
+   */
+  formatDateTime(date: Date, options: Intl.DateTimeFormatOptions = {}): string {
```

Comment:
As you mention, this approach requires people adding a new `date.humanize.datetime` format, which would require translations to connect the date and the time.

Since [INTL.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) provides the concatenation automatically if you specify a date and a time, I suggest instead that you continue to use `formatDate`, but add a new `DateStyle` enum [here](https://github.com/Shopify/quilt/blob/main/packages/react-i18n/src/constants/index.ts#L8), such as `DateStyle.LongDateTime` or `DateStyle.Full`, which would return a `DateStyle.Long` date with a time, like `Thursday, October 15, 2015 at 2:00 PM` in English.

It would be worth testing in different locales, but this might be a good format:
```
  [DateStyle.Full]: {
    weekday: 'long',
    month: 'long',
    day: 'numeric',
    year: 'numeric',
    hour: 'numeric',
    minute: 'numeric',
  },
```

It also looks like this might work, but the first is more explicit:
```
  [DateStyle.Full]: {
    dateStyle: 'full',
    timeStyle: 'short',
  },
```

Reply:
Updated it to be to style below, but as explained on Slack, it is still using a translation. This because the Intl libraries do not concat date and time correctly. They separate it using a comma, while Polaris describes that they should be separated by the word `at`.

In the latest version, it still uses a different translation key as the humanize methods already available in this library.

```
  [DateStyle.LongDateTime]: {
    month: 'short',
    day: 'numeric',
    year: 'numeric',
    hour: 'numeric',
    minute: 'numeric',
  },
```